crates/neon/src/types_impl/promise.rs

```diff
@@ -49,9 +49,74 @@ const BOUNDARY: FailureBoundary = FailureBoundary {
     feature = "promise-api",
     deprecated = "`promise-api` feature has no impact and may be removed"
 )]
-/// The JavaScript [`Promise`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Promise) value.
+/// The type of JavaScript
+/// [`Promise`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Promise)
+/// objects.
 ///
-/// [`JsPromise`] may be constructed with [`Context::promise`].
+/// [`JsPromise`] instances may be constructed with [`Context::promise`].
+///
+/// # Example
+///
+/// ```
+/// # use neon::prelude::*;
+/// fn resolve_promise(mut cx: FunctionContext) -> JsResult<JsPromise> {
+///     let (deferred, promise) = cx.promise();
+///     let msg = cx.string("Hello, World!");
+///
+///     deferred.resolve(&mut cx, msg);
+///
+///     Ok(promise)
+/// }
+/// ```
+///
+/// # Example: Asynchronous task
+///
+/// This example uses the [linkify](https://crates.io/crates/linkify) crate in an
+/// asynchronous task, i.e. a
+/// [Node worker pool](https://nodejs.org/en/docs/guides/dont-block-the-event-loop/)
+/// thread, to find all the links in a text string.
+///
+/// More robust implementations might use a custom Rust thread or thread pool to avoid
+/// blocking the worker pool; for more information, see the [`JsFuture`] example.
+///
+/// ```
+/// # use neon::prelude::*;
+/// use linkify::{LinkFinder, LinkKind};
+/// use easy_cast::Cast; // for safe numerical conversions
+///
+/// fn linkify(mut cx: FunctionContext) -> JsResult<JsPromise> {
+///     let text = cx.argument::<JsString>(0)?.value(&mut cx);
+///
+///     let promise = cx
+///         .task(move || {
+///             let (indices, kinds): (Vec<_>, Vec<_>) = LinkFinder::new()
+///                 .spans(&text)
+///                 .map(|span| {
+///                     let end: u32 = span.end().cast();
+///
+///                     let kind: u8 = match span.kind() {
+///                         Some(LinkKind::Url) => 1,
+///                         Some(LinkKind::Email) => 2,
+///                         _ => 0,
+///                     };
+///
+///                     (end, kind)
+///                 })
+///                 .unzip();
```

Comment:
Unzipping into columnar values is very elegant. It's simple and makes the work on the main thread very light.